.github/workflows/deploy.yml

```diff
@@ -1,39 +1,68 @@
-name: Update GitHub Packages
+name: Update GitHub Packages, deploy to AWS
 
 on:
   push:
+    branches:
+      - main
     tags:
       - v*
 
+env:
+  APP_VERSION: 0.1.0
+  CHART_NAME: simulator
+
 jobs:
-  package:
-    name: Package the app
+  push:
+    name: GitHub Packages Update
     runs-on: ubuntu-latest
     if: github.event_name == 'push'
+    steps:
+      - uses: actions/checkout@v2
 
-    env:
-      GO_VERSION: "1.16.3"
+      - name: Set image tag
+        run: echo "TAG=${GITHUB_REF_NAME}-${APP_VERSION}-${GITHUB_RUN_NUMBER}" >> $GITHUB_ENV
 
-    steps:
-      - name: Checkout project
-        uses: actions/checkout@v2
+      - name: Log into registry
+        run: echo "${{ secrets.GITHUB_TOKEN }}" | docker login ghcr.io -u flapflapio --password-stdin
 
-      - name: Use go${{ env.GO_VERSION }}
-        uses: actions/setup-go@v2
-        with:
-          go-version: ^${{ env.GO_VERSION }}
-      - run: go version
+      - name: Build image
+        run: make docker
 
-      - name: Build and package
-        run: |
-          # TODO: Package up the app
+      - name: Push image
+        run: make docker-push
 
   deploy:
-    name: Update GitHub Packages
+    name: Deploy to AWS
     runs-on: ubuntu-latest
-    needs: package
-
+    needs: push
+    if: github.event_name == 'push'
     steps:
-      - name: Deploy to GitHub packages
+      - uses: actions/checkout@v2
+
+      - name: Create KUBECONFIG
+        run: |
+          echo "${{ secrets.KUBECONFIG_FILE }}" >> kubeconfig.yml
+          chmod 0600 kubeconfig.yml
+          echo "KUBECONFIG=$(realpath kubeconfig.yml)" >> $GITHUB_ENV
+
+      - name: Download Helm
```

Comment:
Why do we need to use HELM in this case?

Reply:
Theoretically, we can just write our YAML resource manifests and deploy them with `kubectl`, but the issue is that we would be hardcoding all the fields in those resources, if we want to change for example the container image tag that we are deploying, then we would need to use something like `sed` in order to modify those YAML files before passing them to kubectl.

Or we could use a templating utility like Python Jinja templates.

But the most commonly used tool for this use case is Helm. The main feature is that templating that I mentioned. We can pass values to Helm and it will insert them into our Kubernetes resource manifests before it applies them.

Helm also gives us some other features like rollback if a deployment fails, and some versioning features.

You can think of helm as basically a package manager for kubernetes. We have written a "helm chart" (i.e. a package deployed by this package manager) for the simulator service and that's how we deploy the app.